controllers/virtinkmachine_controller.go

```diff
@@ -405,13 +413,17 @@ func (r *VirtinkMachineReconciler) buildVM(ctx context.Context, machine *infrast
 }
 
 func (r *VirtinkMachineReconciler) buildDataVolumes(ctx context.Context, machine *infrastructurev1beta1.VirtinkMachine) []*cdiv1beta1.DataVolume {
+	dataVolumeNamespace := machine.Namespace
 	dataVolumes := []*cdiv1beta1.DataVolume{}
 	for _, volume := range machine.Spec.VolumeTemplates {
 		switch {
 		case volume.DataVolume != nil:
+			if volume.DataVolume.ObjectMeta.Namespace != "" {
+				dataVolumeNamespace = volume.DataVolume.ObjectMeta.Namespace
```

Comment:
Maybe we should make sure the namespace of the DataVolume is the same with that of the VM?

Reply:
fixed, now DataVolume use VM's namespace directly.